File: CS/TreeListDrawImage/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraTreeList.ViewInfo;
using DevExpress.XtraTreeList.Nodes;

namespace TreeListDrawImage
{
    public partial class Form1 : Form
    {
        DrawHelper helper;
        public Form1()
        {
            InitializeComponent();
            AddNodes();
            helper = new DrawHelper();
            helper.RegisterTreeList(treeList1, TreeListDrawImage.Properties.Resources.Asphalt_World___16x16);
            helper.ImageClick += helper_ImageClick;
        }

        void helper_ImageClick(object sender, EventArgs e)
        {
            MessageBox.Show("Click!");
        }

        void AddNodes()
        {
            for (int i = 0; i < 10; i++)
            {
                treeList1.AppendNode(new object[] { string.Format("Node {0}", i) }, -1);
            }
        }
        protected override void OnFormClosing(FormClosingEventArgs e) {
            helper.ImageClick -= helper_ImageClick;
            helper.Unregister();
            base.OnFormClosing(e);
        }
    }
}

File: VB/TreeListDrawImage/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList.ViewInfo
Imports DevExpress.XtraTreeList.Nodes

Namespace TreeListDrawImage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			AddNodes()
			Dim helper As New DrawHelper()
			helper.RegisterTreeList(treeList1, My.Resources.Asphalt_World___16x16)
			AddHandler helper.ImageClick, AddressOf helper_ImageClick
		End Sub

		Private Sub helper_ImageClick(ByVal sender As Object, ByVal e As EventArgs)
			MessageBox.Show("Click!")
		End Sub

		Private Sub AddNodes()
			For i As Integer = 0 To 9
				treeList1.AppendNode(New Object() { String.Format("Node {0}", i) }, -1)
			Next i
		End Sub
	End Class
End Namespace
